jupyter_client/utils.py

```diff
@@ -6,17 +6,25 @@
 import asyncio
 import inspect
 import os
+import sys
 
 
 def run_sync(coro):
     def wrapped(*args, **kwargs):
         try:
             loop = asyncio.get_running_loop()
         except RuntimeError:
-            # Workaround for bugs.python.org/issue39529.
-            try:
-                loop = asyncio.get_event_loop_policy().get_event_loop()
-            except RuntimeError:
+            # In python 3.10.9 this workaround breaks some tests
+            # test_client.TestKernelClient
+            # https://github.com/python/cpython/issues/83710
+            if sys.version_info <= (3, 10, 8):
+                # Workaround for bugs.python.org/issue39529.
+                try:
+                    loop = asyncio.get_event_loop_policy().get_event_loop()
+                except RuntimeError:
+                    loop = asyncio.new_event_loop()
+                    asyncio.set_event_loop(loop)
+            else:
```

Comment:
I think this changes the behavior. In the `else`, should we try to use `get_running_loop` before creating a new event loop?